codereview.java

```diff
@@ -0,0 +1,37 @@
+public class RuntimeIssuesDemo {
```

Comment:
code: System.out.println(numbers[3]); // Accessing index 3, which is out of bounds - 
ArrayIndexOutOfBoundsException can be avoided by checking the array length before accessing the index.
code: System.out.println(str.length()); // Attempting to invoke method on null reference - 
NullPointerException can be avoided by checking if the reference is null before invoking methods.
code: int result = 10 / 0; // Division by zero - 
ArithmeticException can be avoided by checking for zero before performing division.
code: int num = Integer.parseInt(numStr); // Parsing a non-numeric string - 
NumberFormatException can be avoided by handling the parsing error using try-catch.
code: String strObj = (String) obj; // Attempting to cast Integer object to String - 
ClassCastException can be avoided by checking the type before casting.
code: names.remove(name); // Modifying the list while iterating over it - 
ConcurrentModificationException can be avoided by using an Iterator and calling remove method on it.